index/test/rtree/rtree_with_strategies.cpp

```diff
@@ -0,0 +1,156 @@
+#include <boost/geometry/geometries/register/box.hpp>
+#include <boost/geometry/geometries/register/segment.hpp>
+
+struct point { double x, y; };
+struct box { point mi, ma; };
+struct segment { point first, second; };
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(point, double, cs::undefined, x, y)
+BOOST_GEOMETRY_REGISTER_BOX(box, point, mi, ma)
+BOOST_GEOMETRY_REGISTER_SEGMENT(segment, point, first, second)
+
+template <typename QueryValue, typename Rtree>
+constexpr bool is_bp_or_bb_v = bg::util::is_box<QueryValue>::value
+                               && bg::util::is_box<typename Rtree::value_type>::value
+                            || bg::util::is_box<QueryValue>::value
+                               && bg::util::is_point<typename Rtree::value_type>::value;
+
+template <typename QueryValue, typename Rtree>
+constexpr bool is_pb_or_bb_v = bg::util::is_box<QueryValue>::value
+                               && bg::util::is_box<typename Rtree::value_type>::value
+                            || bg::util::is_point<QueryValue>::value
+                               && bg::util::is_box<typename Rtree::value_type>::value;
+
+template <typename QueryValue, typename Rtree>
+constexpr bool is_pp_or_bb_v = bg::util::is_box<QueryValue>::value
+                                && bg::util::is_box<typename Rtree::value_type>::value
+                            || bg::util::is_point<QueryValue>::value
+                                && bg::util::is_point<typename Rtree::value_type>::value;
+
+template
+<
+    typename QueryValue, typename Rtree,
+    std::enable_if_t<is_bp_or_bb_v<QueryValue, Rtree>, int> = 0
+>
+void test_queries_bp_bb(Rtree const& rtree)
+{
+    std::vector<typename Rtree::value_type> values;
+    rtree.query(bgi::covered_by(QueryValue{}), std::back_inserter(values));
+    rtree.query(bgi::disjoint(QueryValue{}), std::back_inserter(values));
+    rtree.query(bgi::within(QueryValue{}), std::back_inserter(values));
+}
+
+template
+<
+    typename QueryValue, typename Rtree,
+    std::enable_if_t<(!is_bp_or_bb_v<QueryValue, Rtree>), int> = 0
+>
+void test_queries_bp_bb(Rtree const& ) {}
+
+template
+<
+    typename QueryValue, typename Rtree,
+    std::enable_if_t<is_pb_or_bb_v<QueryValue, Rtree>, int> = 0
+>
+void test_queries_pb_bb(Rtree const& rtree)
+{
+    // These predicates use algorithms that are not implemented for
+    // some geometry combinations
+    std::vector<typename Rtree::value_type> values;
+    rtree.query(bgi::contains(QueryValue{}), std::back_inserter(values));
+    rtree.query(bgi::covers(QueryValue{}), std::back_inserter(values));
+}
+
+template
+<
+    typename QueryValue, typename Rtree,
+    std::enable_if_t<(!is_pb_or_bb_v<QueryValue, Rtree>), int> = 0
+>
+void test_queries_pb_bb(Rtree const& ) {}
+
+template
+<
+    typename QueryValue, typename Rtree,
+    std::enable_if_t<is_pp_or_bb_v<QueryValue, Rtree>, int> = 0
+>
+void test_overlaps_pp_bb(Rtree const& rtree)
```

Comment:
maybe rename to `test_overlaps_touches_pp_bb` ?

Reply:
Right, I forgot to rename it to `test_queries_pp_bb`.